includes/plugins/class-newspack-newsletters.php

```diff
@@ -1,446 +0,0 @@
-				}
-
-				$parsed_url = \wp_parse_url( $raw_url );
-
-				// Maybe set UTM meta.
-				if ( ! empty( $parsed_url['query'] ) ) {
-					$utm_key_prefix = $is_payment ? 'payment_page_utm' : 'signup_page_utm';
-					$params         = [];
-					\wp_parse_str( $parsed_url['query'], $params );
-					foreach ( $params as $param => $value ) {
-						$param = \sanitize_text_field( $param );
-						if ( 'utm' === substr( $param, 0, 3 ) ) {
-							$param = str_replace( 'utm_', '', $param );
-							$key   = self::get_metadata_key( $utm_key_prefix ) . $param;
-							if ( ! isset( $contact['metadata'][ $key ] ) || empty( $contact['metadata'][ $key ] ) ) {
-								$contact['metadata'][ $key ] = $value;
-							}
-						}
-					}
-				}
-			}
-
-			foreach ( $contact['metadata'] as $meta_key => $meta_value ) {
-				if ( self::should_sync_ras_metadata() ) {
-					if ( in_array( $meta_key, $raw_keys, true ) ) {
-						$normalized_metadata[ self::get_metadata_key( $meta_key ) ] = $meta_value; // If passed a raw key, map it to the prefixed key.
-					} elseif (
-						in_array( $meta_key, $prefixed_keys, true ) ||
-						(
-							// UTM meta keys can have arbitrary suffixes.
-							( in_array( self::get_metadata_key( 'signup_page_utm' ), $prefixed_keys, true ) && false !== strpos( $meta_key, self::get_metadata_key( 'signup_page_utm' ) ) ) ||
-							( in_array( self::get_metadata_key( 'payment_page_utm' ), $prefixed_keys, true ) && false !== strpos( $meta_key, self::get_metadata_key( 'payment_page_utm' ) ) )
-						)
-					) {
-						$normalized_metadata[ $meta_key ] = $meta_value;
-					}
-				} else { // phpcs:ignore Universal.ControlStructures.DisallowLonelyIf.Found
-					// If not syncing for RAS, we only want to sync email (for all ESPs) + First/Last Name (for MC only).
-					if ( in_array( $meta_key, [ 'First Name', 'Last Name' ], true ) ) {
-						$normalized_metadata[ $meta_key ] = $meta_value;
-					}
-				}
-			}
-
-			// Ensure status is passed, if given.
-			if ( isset( $contact['metadata']['status'] ) ) {
-				$normalized_metadata['status'] = $contact['metadata']['status'];
-			}
-			if ( isset( $contact['metadata']['status_if_new'] ) ) {
-				$normalized_metadata['status_if_new'] = $contact['metadata']['status_if_new'];
-			}
-
-			$contact['metadata'] = $normalized_metadata;
-		}
-
-		// Parse full name into first + last for MC, which stores these as separate merge fields.
-		if ( 'mailchimp' === \get_option( 'newspack_newsletters_service_provider', false ) ) {
-			if ( isset( $contact['name'] ) ) {
-				if ( ! isset( $contact['metadata'] ) ) {
-					$contact['metadata'] = [];
-				}
-				$name_fragments                    = explode( ' ', $contact['name'], 2 );
-				$contact['metadata']['First Name'] = $name_fragments[0];
-				if ( isset( $name_fragments[1] ) ) {
-					$contact['metadata']['Last Name'] = $name_fragments[1];
-				}
-			}
-		}
```

Comment:
This should now live here: https://github.com/Automattic/newspack-newsletters/pull/1628